docs/tutorial.md

```diff
@@ -0,0 +1,372 @@
+[quay.io](https://quay.io/)) and be logged in in your command line environment.
+  - `example.com` is used as the registry Docker Hub namespace in these examples.
+  Replace it with another value if using a different registry or namespace.
+  - [Authentication and certificates][image-reg-config] if the registry is private or uses a custom CA.
+
+  ## Overview
+
+We will create a sample project to let you know how it works and this sample will:
+
+- Create a Memcached Deployment if it doesn't exist
+- Ensure that the Deployment size is the same as specified by the Memcached CR spec
+- Update the Memcached CR status using the status writer with the names of the CR's pods
+
+## Create a new project
+
+Use the CLI to create a new memcached-operator project:
+
+```sh
+mkdir -p $HOME/projects/memcached-quarkus-operator
+cd $HOME/projects/memcached-quarkus-operator
+# we'll use a domain of example.com
+# so all API groups will be <group>.example.com
+operator-sdk init --plugins quarkus --domain lucky.com --project-name java-op
+```
+
+#### A note on dependency management
+
+`operator-sdk init` generates `pom.xml` file. This file contains all the depedencies required to run the operator.
+
+### JavaOpOperator
+
+The main program of the operator `JavaOpOperator.java` initializes and runs the operator. The operator uses java-operator-sdk which is similar to Go lang version of controller-runtime.
+
+Below code will initize and defines the informers/watches for your operator.
+
+```
+  @Override
+  public int run(String... args) throws Exception {
+    operator.start();
+
+    Quarkus.waitForExit();
+    return 0;
+  }
+```
+
+## Create a new API and Controller
+
+Create a new Custom Resource Definition (CRD) API with group `cache` version `v1alpha1` and Kind Memcached. Below messages are informational mesages.
+
+```console
+$ operator-sdk create api --plugins quarkus --group cache --version v1 --kind Memcached
+InjectResource called
+UpdateResource called
+Scaffold called
+NewCreateAPIScaffolder called
+InjectFS called
+api.Scaffold()
+PostScaffold called
+...
+```
+
+This will scaffold the MemcachedController, MemcachedSpec, MemcachedStatus and Memcached.
+
+#### Understanding Kubernetes APIs
+
+For an in-depth explanation of Kubernetes APIs and the group-version-kind model, check out these [kubebuilder docs][kb-doc-gkvs].
+
+The API's used from [java-operator-sdk][https://github.com/java-operator-sdk/java-operator-sdk]. It is similar to `Controller-runtime` from Go lang.
+
+### Define the API
+
+To begin, we will represent our API by defining the `Memcached` type, which will have a `MemcachedSpec.Size` field to set the quantity of memcached instances (CRs) to be deployed, and a `MemcachedStatus.Nodes` field to store a CR's Pod names.
+
+**Note** The Node field is just to illustrate an example of a Status field. In real cases, it would be recommended to use [Conditions][conditionals].
```

Comment:
where is the `conditionals` url defined?

Reply:
Modified.